cmd/erasure-object.go

```diff
@@ -1915,17 +1915,20 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 	versionFound := true
 	objInfo = ObjectInfo{VersionID: opts.VersionID} // version id needed in Delete API response.
 	goi, _, gerr := er.getObjectInfoAndQuorum(ctx, bucket, object, opts)
+	tryDel := false
 	if gerr != nil && goi.Name == "" {
 		if _, ok := gerr.(InsufficientReadQuorum); ok {
 			// Add an MRF heal for next time.
 			er.addPartial(bucket, object, opts.VersionID)
-
-			return objInfo, InsufficientWriteQuorum{}
+			if opts.Versioned || opts.VersionSuspended {
+				return objInfo, InsufficientWriteQuorum{}
+			}
+			tryDel = true // only for unversioned objects
```

Comment:
if the "version" is specified, can we do a permanent purge by marking xl.meta with "versionPurgeStatus" ?

Reply:
versionPurgeStatus is relevant only for replication because it needs to maintain state. If there is sufficient quorum and metadata can be read, versioned delete can be attempted for non replication cases - perhaps we can check if suff disks online before DeleteVersion?

Reply:
Yeah definitely. My point was if delete was issued permanently we can update metadata forcibly?

Reply:
my only concern is on whether bucket metadata is correctly loaded at the time of delete - as long as that is known, permanent delete can be forced through for regular versioned objects (barring object lock/replication)